hack/tools/tilt-prepare/main.go

```diff
@@ -0,0 +1,549 @@
+		if err != nil {
+			return errors.Wrapf(err, "[%s] failed to read existing %s", prefix, path)
+		}
+		if bytes.Equal(previousYaml, yaml) {
+			klog.Infof("[%s] no changes in the generated yaml\n", prefix)
+			return nil
+		}
+	}
+
+	err = os.MkdirAll(filepath.Dir(path), 0750)
+	if err != nil {
+		return errors.Wrapf(err, "[%s] failed to create dir %s", prefix, filepath.Dir(path))
+	}
+
+	if err := os.WriteFile(path, yaml, 0600); err != nil {
+		return errors.Wrapf(err, "[%s] failed to write %s", prefix, path)
+	}
+	return nil
+}
+
+// prepareDeploymentForDebug alter controller deployments for working nicely with delve debugger;
+// most specifically, liveness and readiness probes are dropper and leader election turned off.
+func prepareDeploymentForDebug(prefix string, objs []unstructured.Unstructured) error {
+	for i := range objs {
+		obj := objs[i]
+		if obj.GetKind() != "Deployment" {
+			continue
+		}
+
+		// Convert Unstructured into a typed object
+		d := &appsv1.Deployment{}
+		if err := scheme.Scheme.Convert(&obj, d, nil); err != nil {
+			return errors.Wrapf(err, "[%s] failed to convert Deployment to typed object", prefix)
+		}
+
+		for j := range d.Spec.Template.Spec.Containers {
+			container := d.Spec.Template.Spec.Containers[j]
+
+			// Drop liveness and readiness probes.
+			container.LivenessProbe = nil
+			container.ReadinessProbe = nil
+
+			// Drop leader election.
+			debugArgs := make([]string, 0, len(container.Args))
+			for _, a := range container.Args {
+				if a == "--leader-elect" || a == "--leader-elect=true" {
+					continue
+				}
+				debugArgs = append(debugArgs, a)
+			}
+			container.Args = debugArgs
+
+			d.Spec.Template.Spec.Containers[j] = container
+		}
+
+		if err := scheme.Scheme.Convert(d, &obj, nil); err != nil {
+			return errors.Wrapf(err, "[%s] failed to convert Deployment to unstructured", prefix)
+		}
+
+		objs[i] = obj
+	}
+	return nil
+}
+
+func getProviderObj(prefix string, objs []unstructured.Unstructured) (*unstructured.Unstructured, error) {
+	namespace := ""
+	manifestLabel := ""
+	for i := range objs {
+		if objs[i].GetKind() != "Namespace" {
+			continue
+		}
+
+		namespace = objs[i].GetName()
+		manifestLabel = objs[i].GetLabels()[clusterv1.ProviderLabelName]
+		break
+	}
+
+	providerType := "CoreProvider"
+	providerName := manifestLabel
+	if strings.HasPrefix(manifestLabel, "infrastructure-") {
+		providerType = "InfrastructureProvider"
+		providerName = manifestLabel[len("infrastructure-"):]
+	}
+	if strings.HasPrefix(manifestLabel, "bootstrap-") {
+		providerType = "BootstrapProvider"
+		providerName = manifestLabel[len("bootstrap-"):]
+	}
+	if strings.HasPrefix(manifestLabel, "control-plane-") {
+		providerType = "ControlPlaneProvider"
+		providerName = manifestLabel[len("control-plane-"):]
+	}
+
+	provider := &clusterctlv1.Provider{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Provider",
+			APIVersion: clusterctlv1.GroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
```

Comment:
Missed that one (I cannot create a new comment at a line where there is already an old one)
```go
Name: manifestLabel
```

Reply:
fixed